dialog/file.go

```diff
@@ -233,7 +233,33 @@ func (f *fileDialog) makeUI() fyne.CanvasObject {
 		}
 	})
 
+	newFolderButton := widget.NewButtonWithIcon("", theme.FolderNewIcon(), func() {
+		newFolderEntry := widget.NewEntry()
+		ShowForm("New Folder", "Create Folder", "Cancel", []*widget.FormItem{
+			{
+				Text:   "Name",
+				Widget: newFolderEntry,
+			},
+		}, func(s bool) {
+			if !s || newFolderEntry.Text == "" {
+				return
+			}
+
+			newFolderPath := filepath.Join(f.dir.Path(), newFolderEntry.Text)
+			createFolderErr := os.MkdirAll(newFolderPath, 0000)
```

Comment:
The permission values do not seem correct to me. I think you likely want to use 0700 instead.

Reply:
Sorry to be a pain, but shouldn't it be 0733? These folders are not restricted to user home folders so they could be writing files that other users should see...

Reply:
thanks for pointing that out. actually i would think that it should be 0755? we would like the newly created folders to be seen by others (via `ls`) and that others can `cd` into those directories. these corresponds to `r` and `x` respectively. what we might not want is for others to modify the directory's contents (e.g. renaming of subfiles), which corresponds to `w`. `r-x` corresponds to 0755.

honestly, it might be better to have permissions set to 0777 since like you pointed out, they might be creating folders for others to write to as well, and fyne shouldn't restrict on that?

not sure if i grasped the concepts correctly but these are my thoughts. thanks a bunch !

Reply:
You're quite right - I meant 755. However you could be on to something with 777. If the system umask applies to this it would make complete sense. What do you think @Jacalz?

Reply:
Hmm. I had a look at what `gosec` is recommending and I think they don’t want anything higher than 0750. Will have a closer look at that later today.

Reply:
I ran the numbers through a Unix permission calculator and I think 0750 makes sense. That means read/write/execute for user and read/execute for group with none for other.

Reply:
agree with that reasoning